backend/isoko-backend/src/handlers/review/post-review.js

```diff
@@ -0,0 +1,68 @@
+const dynamodb = require('aws-sdk/clients/dynamodb');
+const docClient = new dynamodb.DocumentClient();
+const _ = require('lodash');
+const { BUSINESS_TABLE } = require('../../constants');
+
+/**
+ * HTTP post method that creates a review for the specified business.
+ */
+exports.postReviewHandler = async (event) => {
+   if (event.httpMethod !== 'POST') {
+      throw new Error(
+         `postReviewHandler only accept POST method, you tried: ${event.httpMethod}`
+      );
+   }
+
+   console.info('received:', event);
+
+   const { businessId } = event.pathParameters;
+
+   if (businessId == null) {
+      throw new Error(
+         `Missing query parameter 'businessId'. Request URL format: POST/business/:businessId/review`
+      );
+   }
+
+   const requestBody = event.body && JSON.parse(event.body);
+
+   const reviewAuthor = _.get(requestBody, 'reviewAuthor');
+   const authorUserName = _.get(requestBody, 'authorUserName');
+   const authorProfilePicture = _.get(requestBody, 'authorProfilePicture', '');
+   const stars = _.get(requestBody, 'stars');
+   const reviewTitle = _.get(requestBody, 'reviewTitle', '');
+   const description = _.get(requestBody, 'description', '');
+   const pictures = _.get(requestBody, 'pictures', []);
+   const ts = _.get(requestBody, 'ts');
+
+   const params = {
+      TableName: BUSINESS_TABLE,
+      Item: {
+         pk: businessId,
+         sk: 'REVIEW#' + ts + '#' + authorUserName,
+         reviewAuthor: reviewAuthor,
+         authorUserName: authorUserName,
+         authorProfilePicture: authorProfilePicture,
+         stars: stars,
+         reviewTitle: reviewTitle,
+         description: description,
+         pictures: pictures,
+         ts: ts,
+      },
+   };
+
+   const dynamoResult = await docClient.put(params).promise();
```

Comment:
This is something I just realized we should be doing on all of our endpoints, but we should catch any error thrown by this call and return it in a 400 response to the user if it does. I think right now since we don't catch an error it will throw a 500 response to the user

Reply:
Seems like dynamodb should already handle this if I'm understanding this properly: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html

Reply:
Yeah so you're right that Dynamo will throw a 400 response, but we don't then send that back to the person who sent our API the request. Also since 400 means an error I'm pretty sure it just rejects the promise throwing an error and since we don't catch it it breaks our code. When that happens we send a 500 response to the requester which means like our API is broken, but we should be sending the error response dynamo gave us instead.

Reply:
I can explain more in class tomorrow if that doesn't make sense